src/apps/explorer/pages/Metadata/config.ts

```diff
@@ -0,0 +1,60 @@
+import { UiSchema } from '@rjsf/core'
+import { JSONSchema7 } from 'json-schema'
+import { AjvError, FormValidation } from '@rjsf/core'
+import { isAddress } from 'web3-utils'
+
+export const schema: JSONSchema7 = {
+  title: 'Metadata Form',
+  type: 'object',
+  required: ['version'],
+  properties: {
+    appCode: { type: 'string', title: 'AppCode' },
+    version: { type: 'string', title: 'Version', readOnly: true, default: '1.0.0' },
+    environment: { type: 'string', title: 'Environment' },
+    metadata: {
+      type: 'object',
+      title: 'Metadata',
+      properties: {
+        referrer: {
+          type: 'object',
+          title: 'Referrer',
+          properties: {
+            version: { type: 'string', title: 'Version', readOnly: true, default: '1.0.0' },
+            address: { type: 'string', title: 'Address' },
+          },
+          required: ['version'],
+        },
+        quote: {
+          type: 'object',
+          title: 'Quote',
+          properties: {
+            version: { type: 'string', title: 'Quote' },
+            slippageBips: { type: 'string', title: 'Slippage Bips', pattern: '^\\d*$' },
+          },
+        },
+      },
+    },
+  },
+}
```

Comment:
shouldn't this be loaded through the sdk or the app-data package directly? @alfetopito I believe the json schema should be exposed through the sdk right? In that case maybe @matextrem can work locally with the app-data changes on the sdk before you publish version 1.0.0, wdyt?

Reply:
The SDK changes in regards to app-data are still pending review https://github.com/cowprotocol/cow-sdk/pull/50

Still, it is not exposing the appData schemas directly, only using it internally.
Do you think it makes sense to re-expose the json schemas as well?

Reply:
For this case exposing the compiled schema as in `@cowprotocol/app-data/schemas/v0.4.0.json` should allow the react-jsonschema-form lib to create dynamically the form without defining it manually as @matextrem is doing here.

I tested the the latest compiled schema in react-jsonschema-form's [playground and works great](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJtZXRhZGF0YSI6eyJyZWZlcnJlciI6e30sInF1b3RlIjp7fX0sImZpcnN0TmFtZSI6IkNodWNrIiwibGFzdE5hbWUiOiJOb3JyaXMiLCJhZ2UiOjc1LCJiaW8iOiJSb3VuZGhvdXNlIGtpY2tpbmcgYXNzZXMgc2luY2UgMTk0MCIsInBhc3N3b3JkIjoibm9uZWVkIn0sInNjaGVtYSI6eyIkaWQiOiJodHRwczovL2Nvd3N3YXAuZXhjaGFuZ2Uvc2NoZW1hcy9hcHAtZGF0YS92MC40LjAuanNvbiIsIiRzY2hlbWEiOiJodHRwOi8vanNvbi1zY2hlbWEub3JnL2RyYWZ0LTA3L3NjaGVtYSIsImRlc2NyaXB0aW9uIjoiTWV0YWRhdGEgSlNPTiBkb2N1bWVudCBmb3IgYWRkaW5nIGluZm9ybWF0aW9uIHRvIG9yZGVycy4iLCJyZXF1aXJlZCI6WyJ2ZXJzaW9uIiwibWV0YWRhdGEiXSwidGl0bGUiOiJBcHBEYXRhIFJvb3QgU2NoZW1hIiwidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsidmVyc2lvbiI6eyIkaWQiOiIjL2RlZmluaXRpb25zL3ZlcnNpb24iLCJkZXNjcmlwdGlvbiI6IlNlbWFudGljIHZlcnNpb25pbmcgb2YgZG9jdW1lbnQiLCJleGFtcGxlcyI6WyIxLjAuMCIsIjEuMi4zIl0sInRpdGxlIjoiVmVyc2lvbiIsInR5cGUiOiJzdHJpbmcifSwiYXBwQ29kZSI6eyIkaWQiOiIjL3Byb3BlcnRpZXMvYXBwQ29kZSIsImRlc2NyaXB0aW9uIjoiVGhlIGNvZGUgaWRlbnRpZnlpbmcgdGhlIENMSSwgVUksIHNlcnZpY2UgZ2VuZXJhdGluZyB0aGUgb3JkZXIuIiwiZXhhbXBsZXMiOlsiQ293U3dhcCJdLCJ0aXRsZSI6IkFwcCBDb2RlIiwidHlwZSI6InN0cmluZyJ9LCJlbnZpcm9ubWVudCI6eyIkaWQiOiIjL3Byb3BlcnRpZXMvZW52aXJvbm1lbnQiLCJkZXNjcmlwdGlvbiI6IkVudmlyb25tZW50IGZyb20gd2hpY2ggdGhlIG9yZGVyIGNhbWUgZnJvbSIsInRpdGxlIjoiRW52aXJvbm1lbnQiLCJ0eXBlIjoic3RyaW5nIiwiZXhhbXBsZXMiOlsicHJvZHVjdGlvbiIsImRldmVsb3BtZW50Iiwic3RhZ2luZyIsImVucyJdfSwibWV0YWRhdGEiOnsiJGlkIjoiIy9wcm9wZXJ0aWVzL21ldGFkYXRhIiwiZGVmYXVsdCI6e30sImRlc2NyaXB0aW9uIjoiRWFjaCBtZXRhZGF0YSB3aWxsIHNwZWNpZnkgb25lIGFzcGVjdCBvZiB0aGUgb3JkZXIuIiwicmVxdWlyZWQiOltdLCJ0aXRsZSI6Ik1ldGFkYXRhIiwidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsicmVmZXJyZXIiOnsiJGlkIjoiI3JlZmVycmVyL3YwLjEuMC5qc29uIiwiJHNjaGVtYSI6Imh0dHA6Ly9qc29uLXNjaGVtYS5vcmcvZHJhZnQtMDcvc2NoZW1hIiwicmVxdWlyZWQiOlsidmVyc2lvbiIsImFkZHJlc3MiXSwidGl0bGUiOiJSZWZlcnJlciIsInR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7InZlcnNpb24iOnsiJHJlZiI6IiMvcHJvcGVydGllcy92ZXJzaW9uIn0sImFkZHJlc3MiOnsidGl0bGUiOiJSZWZlcnJlciBhZGRyZXNzIiwiJGlkIjoiIy9kZWZpbml0aW9ucy9ldGhlcmV1bUFkZHJlc3MiLCJwYXR0ZXJuIjoiXjB4W2EtZkEtRjAtOV17NDB9JCIsImV4YW1wbGVzIjpbIjB4YjZCQWQ0MWFlNzZBMTFEMTBmN2IwRTY2NEM1MDA3YjkwOGJDNzdDOSJdLCJ0eXBlIjoic3RyaW5nIn19fSwicXVvdGUiOnsiJGlkIjoiI3F1b3RlL3YwLjIuMC5qc29uIiwiJHNjaGVtYSI6Imh0dHA6Ly9qc29uLXNjaGVtYS5vcmcvZHJhZnQtMDcvc2NoZW1hIiwicmVxdWlyZWQiOlsidmVyc2lvbiIsInNsaXBwYWdlQmlwcyJdLCJ0aXRsZSI6IlF1b3RlIiwidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsidmVyc2lvbiI6eyIkcmVmIjoiIy9wcm9wZXJ0aWVzL3ZlcnNpb24ifSwic2xpcHBhZ2VCaXBzIjp7IiRpZCI6IiMvcHJvcGVydGllcy9zbGlwcGFnZUJpcHMiLCJ0aXRsZSI6IlNsaXBwYWdlIEJpcHMiLCJkZXNjcmlwdGlvbiI6IlNsaXBwYWdlIHRvbGVyYW5jZSB0aGF0IHdhcyBhcHBsaWVkIHRvIHRoZSBvcmRlciB0byBnZXQgdGhlIGxpbWl0IHByaWNlLiBFeHByZXNzZWQgaW4gQmFzaXMgUG9pbnRzIChCSVBTKSIsImV4YW1wbGVzIjpbIjUiLCIxMCIsIjIwIiwiMTAwIl0sInBhdHRlcm4iOiJeXFxkKyhcXC5cXGQrKT8kIiwidHlwZSI6InN0cmluZyJ9fX19fX19LCJ1aVNjaGVtYSI6eyJmaXJzdE5hbWUiOnsidWk6YXV0b2ZvY3VzIjp0cnVlLCJ1aTplbXB0eVZhbHVlIjoiIiwidWk6YXV0b2NvbXBsZXRlIjoiZmFtaWx5LW5hbWUifSwibGFzdE5hbWUiOnsidWk6ZW1wdHlWYWx1ZSI6IiIsInVpOmF1dG9jb21wbGV0ZSI6ImdpdmVuLW5hbWUifSwiYWdlIjp7InVpOndpZGdldCI6InVwZG93biIsInVpOnRpdGxlIjoiQWdlIG9mIHBlcnNvbiIsInVpOmRlc2NyaXB0aW9uIjoiKGVhcnRoaWFuIHllYXIpIn0sImJpbyI6eyJ1aTp3aWRnZXQiOiJ0ZXh0YXJlYSJ9LCJwYXNzd29yZCI6eyJ1aTp3aWRnZXQiOiJwYXNzd29yZCIsInVpOmhlbHAiOiJIaW50OiBNYWtlIGl0IHN0cm9uZyEifSwiZGF0ZSI6eyJ1aTp3aWRnZXQiOiJhbHQtZGF0ZXRpbWUifSwidGVsZXBob25lIjp7InVpOm9wdGlvbnMiOnsiaW5wdXRUeXBlIjoidGVsIn19fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZSI6ZmFsc2UsInJlYWRvbmx5IjpmYWxzZSwib21pdEV4dHJhRGF0YSI6ZmFsc2UsImxpdmVPbWl0IjpmYWxzZX19)

We should be exposing an utility method in the [Metadata API](https://github.com/cowprotocol/cow-sdk/blob/main/src/api/metadata/index.ts) like `getSchema(version: string = LATEST_APP_DATA_VERSION)`

Reply:
btw, was revisiting the main CowSdk file and we still have there  the [`validateAppDataDocument`](https://github.com/cowprotocol/cow-sdk/blob/main/src/CowSdk.ts#L42) method. Shouldn't we move that to the [Metadata API](https://github.com/cowprotocol/cow-sdk/blob/main/src/api/metadata/index.ts)?

Reply:
Hmm, true. There is definitely more stuff to be hoisted up from the SDK to app-data package and exposed via API methods.
Let's discuss more in detail during the sync meeting

Reply:
@alfetopito Is it possible to change schema to looks like the one I have in` config.tsx` ? Since we need to add some dependencies and some other keys in order for us to allow enable/disable metadata fields

Reply:
what if if we just decorate the json schema with the props that we need? I don't think updating the original schema with UI stuff is a good approach as it will impact all consumers

Reply:
I second Ramiro. Having anything else in the schema will signal that it could be included in the appData, which is not what we want.
If extra data is needed it can be iterated over and enhanced here, but it does not belong in the exported schema definitions

Reply:
New methods on SDK https://github.com/cowprotocol/cow-sdk/pull/54